ICSharpCode.Decompiler/IL/Transforms/InlineExpressionTreeParameterDeclarations.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Diagnostics;
+using System.Linq;
+using ICSharpCode.Decompiler.TypeSystem;
+
+namespace ICSharpCode.Decompiler.IL.Transforms
+{
+	class InlineExpressionTreeParameterDeclarationsTransform : IStatementTransform
+	{
+		public void Run(Block block, int pos, StatementTransformContext context)
+		{
+			if (block.Instructions[pos] is StLoc parameterVariableInst) {
+				bool alwaysInlined = true;
+				if (MatchParameterVariableAssignment(parameterVariableInst, out ILVariable v, out ILInstruction init)) {
+					foreach (var LdParameterVariableInst in context.Function.Descendants.OfType<LdLoc>()) {
+						if (LdParameterVariableInst.MatchLdLoc(v)) {
+							ILInstruction parent = LdParameterVariableInst.Parent;
```

Comment:
You could extract the inner loop into its own (local) method and get rid of the goto. That would make it easier to read.